citrination_client/models/design/constraints/tests/test_constraints.py

```diff
@@ -54,6 +67,47 @@ def test_real_range_constraint_validation():
     except CitrinationClientError:
         pass
 
+
+def test_int_range_constraint():
+
+    prop = "Ingredient count"
+    c = IntRangeConstraint(name=prop, minimum=1, maximum=2)
+    mapped_c_range = c.to_dict()
+
+    assert mapped_c_range["name"] == prop
+    assert mapped_c_range["type"] == "integer"
+    assert mapped_c_range["options"].get("min") == 1
+    assert mapped_c_range["options"].get("max") == 2
+
+def test_int_range_constraint_validation():
+    """
+    Tests that minimum and maximum values are validated correctly
+    on initialization of IntRangeConstraint
+    """
+
+    # Test valid values OK
+    minimum = 1
+    maximum = 2
+    IntRangeConstraint(name="Ingredient count", minimum=minimum, maximum=maximum)
+
+    # Test minimum must be less than maximum
+    minimum = 3
+    maximum = 2
+    try:
+        RealRangeConstraint(name="Ingredient count", minimum=minimum, maximum=maximum)
+        assert False, "IntRangeConstraint should require that minimum be less than maximum"
+    except CitrinationClientError:
+        pass
+
+    # Test values must be castable to float
+    minimum = {}
+    maximum = 2
+    try:
```

Comment:
FYI py.test has a pattern for asserting exceptions are raised: https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions